x-pack/plugins/endpoint/server/routes/resolver/utils/tree.ts

```diff
@@ -50,181 +58,101 @@ function createNode(id: string): ResolverNode {
  * what this means is that noisy neighbors for a given level may hide other child process events that occur later
  * temporally in the same level--so, while a heavily forking process might get shown, maybe the actually malicious
  * event doesn't show up in the tree at the beginning.
+ *
+ * This Tree's root/origin could be in the middle of the tree. The origin corresponds to the id passed in when this
+ * Tree object is constructed. The tree can have ancestors and children coming from the origin.
  */
-
 export class Tree {
-  protected cache: Map<string, ResolverNode>;
-  protected root: ResolverNode;
-  protected id: string;
-
-  constructor(id: string) {
-    const root = createNode(id);
-    this.id = id;
-    this.cache = new Map();
-    this.root = root;
-    this.cache.set(id, root);
+  protected cache: Map<string, Node> = new Map();
```

Comment:
The Tree class is just cumulative container for all the individual handler's responses now.